backend/server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/push/parsing/RuleSetChangedEventParser.java

```diff
@@ -49,7 +50,9 @@ public Optional<RuleSetChangedEvent> parse(String jsonData) {
         changedRule.language,
         IssueSeverity.valueOf(changedRule.severity),
         changedRule.params.stream().filter(p -> p.value != null).collect(Collectors.toMap(p -> p.key, p -> p.value)),
-        changedRule.templateKey))
+        changedRule.templateKey,
+        changedRule.impacts.stream().map(impact -> new OverriddenImpact(impact.getSoftwareQuality(), impact.getSeverity())).collect(Collectors.toList())
```

Comment:
Be careful, on older SonarQubes the `impacts` field will be null since it is not sent in the event. You should have a check here, and either propagate the null `impacts` to `RuleSetChangedEvent`, or make it an empty list (which is maybe simpler)